anchor/src/pubkey-protocol-exports.ts

```diff
@@ -63,22 +63,27 @@ export function getPubKeyPointerPda({
     Uint8Array.from([
       ...PUBKEY_PROTOCOL_PREFIX,
       ...PUBKEY_PROTOCOL_SEED_POINTER,
-      ...stringToUint8Array(provider),
+      ...stringToUint8Array(provider.toString()),
       ...stringToUint8Array(providerId),
     ]),
   )
 
   return PublicKey.findProgramAddressSync([hash], programId)
 }
 
-export enum PubKeyIdentityProvider {
-  Discord = 'Discord',
-  Github = 'Github',
-  Google = 'Google',
-  Solana = 'Solana',
-  X = 'X',
+export const pubKeyIdentityProvider = {
```

Comment:
Let's keep this an enum so it's easier to work with on the frontend.

Reply:
Have refactored these and put a hefty comment in to understand how enums work between AnchorSerialize and generated Typescript types